### PR TITLE
Aruha 1313

### PR DESCRIPTION
**Pull request by v-stepanov**
[ARUHA-1313](https://techjira.zalando.net/browse/ARUHA-1313)

## Description
Fixed subscription /stats endpoint

## Review
- [x] Tests
- [ ] Documentation
- [x] CHANGELOG

## Deployment Notes
Deploy and enjoy

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionNode.java [hunk 1]
```diff
@@ -39,23 +38,24 @@ public void setSessions(final Session[] sessions) {
...
                 .findAny();
     }
 
     @Nullable
-    public String guessStream(final String partition) {
-        return getPartitionWithActiveSession(partition).map(Partition::getSession).orElse(null);
+    public String guessStream(final String eventType, final String partition) {
+        return getPartitionWithActiveSession(eventType, partition)
+                .map(Partition::getSession)
+                .orElse(null);
```
Comment (adyach) [inline src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionNode.java]: Could you return empty string here? In that case you won't need Optional at SubscriptionService.java:270

File: src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionNode.java [hunk 2]
```diff
@@ -39,23 +38,24 @@ public void setSessions(final Session[] sessions) {
...
+        return getPartitionWithActiveSession(eventType, partition)
+                .map(Partition::getState)
+                .orElse(Partition.State.UNASSIGNED);
     }
 
-    private Optional<Partition> getPartitionWithActiveSession(final String partition) {
+    private Optional<Partition> getPartitionWithActiveSession(final String eventType, final String partition) {
         return Stream.of(partitions)
-                .filter(p -> p.getPartition().equals(partition))
+                .filter(p -> p.getPartition().equals(partition) && p.getEventType().equals(eventType))
```
[the remark this answers is not in the document]
Comment (v-stepanov) [reply src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionNode.java]: yes

File: src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionNode.java (hunk 1 above)
Comment (v-stepanov) [reply src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionNode.java]: For me this is actually an anti-pattern to return empty String when there is no value. That's why I would not like to change it. We replace `null` with empty String before showing that to user - which is more or less fine. But I think this method should not return empty String to indicate no value.

Comment (adyach) [reply src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionNode.java]: Returning `null` is even worse! Return optional then :)

Comment (v-stepanov) [reply src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionNode.java]: No, I think returning `null` is much better than returning empty String to indicate that there is no value.
Also, the purpose of my PR was a bugfix, not a refactoring. I don't really understand why I should refactor here :)

**Comment by adyach** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Comment by v-stepanov** [comment]
deploy validation please

**Comment by lmontrieux** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Pull request merged**